### ISSUE TITLE
reject() missing in Option, Try and Either

### ISSUE THREAD

**Issue by Abnaxos** [vavr-io/vavr#2402]
Just had to write this:

```java
Some(myString).filter(s -> !s.isEmpty()).getOrElse("Default")
```

instead of

```java
Some(myString).reject(String::isEmpty).getOrElse("Default")
```

Same for `Try` and `Either`.

**Comment by danieldietrich** [comment]
You are right! In Scala `reject` is called `filterNot`. Let's do it!

**Comment by danieldietrich** [comment]
Interestingly, Traversable#reject(Predicate) exists instead of #filterNot in VAVR... In Scala, collections also have filterNot.

**Comment by danieldietrich** [comment]
We should deprecate _reject_ and add _filterNot_ in all places. This is also a preparation for 1.0.0.

**Comment by Abnaxos** [comment]
I personally prefer *reject*, but whatever. That's a matter of taste. I was just expecting it in Option because it exists in Traversable, but it wasn't there …

*filterNot* has certainly the advantage that it's easier to find: *fil*&lt;Ctrl-Space&gt; – ah, there's a *filterNot*.

**Comment by sraaphorst** [comment]
I'm new, but I would like to take a crack at this PR in order to begin participating in vavr.

**Comment by sraaphorst** [comment]
I'd also like to see `nonEmpty` instead of just `empty`; I've found this quite handy in Scala at times

**Comment by danieldietrich** [comment]
@sraaphorst Traversable [already has](https://static.javadoc.io/io.vavr/vavr/0.10.0/io/vavr/collection/Traversable.html#nonEmpty--) nonEmpty().

We align tightly to Scala. Option et al do have their own dedicated API, e.g.

Option --> isEmpty()
Either --> isLeft()
Try --> isFailure()

**Comment by danieldietrich** [comment]
Merged PR #2414
